snapshot-service/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+mod handlers;
+mod routes;
+
+use jormungandr_lib::{crypto::account::Identifier, interfaces::Value};
+pub use routes::filter;
+use sled::Transactional;
+use std::cmp::max;
+use voting_hir::VoterHIR;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    DbError(#[from] sled::Error),
+
+    #[error(transparent)]
+    DbTxError(#[from] sled::transaction::TransactionError),
+
+    #[error("internal error")]
+    InternalError,
+}
+
+pub type Tag = String;
+type Group = String;
+
+#[derive(Clone)]
+pub struct SharedContext {
+    db: sled::Db,
+}
+
+impl SharedContext {
+    #[tracing::instrument(skip(self))]
+    pub fn get_voting_power(
+        &self,
+        tag: &str,
+        id: &Identifier,
+    ) -> Result<Option<Vec<(Group, Value)>>, Error> {
+        let entries = self.db.open_tree("entries")?;
+        let tags = self.db.open_tree("tags")?;
+
+        let tag = if let Some(tag) = tags.get(tag)? {
+            tag
+        } else {
+            return Ok(None);
+        };
+
+        let key = {
+            let mut key = [0u8; 32 + 4];
+
+            let (tag_part, id_part) = key.split_at_mut(4);
+            tag_part.copy_from_slice(&*tag);
+
+            id_part.copy_from_slice(id.as_ref().as_ref());
+
+            key
+        };
+
+        let key_len = key.len();
+
+        let mut result: Vec<(Group, Value)> = vec![];
+
+        for entries in entries.range(key.clone()..) {
+            let (k, v) = entries?;
+
+            if k[0..key_len] > key[..] {
```

Comment:
I guess it's because you encode groups in the key as well, but this is a bit cryptic without explanation.

Also it could panic if `k.len() < key_len`, but not sure if we want to check this